### PR TITLE
add latest tag for image name which dosen't have tag name

### PR DESCRIPTION
**Pull request by xiaomuqiao**
fix issue about pulling all tags when  image name which dosen't have tag name

### CODE REVIEW

**Comment by xiaomuqiao** [comment]
PTAL

File: docker/manager.go [hunk 1]
```diff
@@ -85,6 +86,16 @@ func NewManager(endpoint string, certPath string, registry api.RegistryCompose)
 
 // PullImage pulls an image by its name.
 func (dm *Manager) PullImage(imageName string) error {
+	repo, err := reference.Parse(imageName)
+	if err != nil {
+		log.Errorf("imagename parse error: %v", err)
```
Comment (superxi911) [inline docker/manager.go]: Why not return?

File: docker/manager.go [hunk 2]
```diff
@@ -85,6 +86,16 @@ func NewManager(endpoint string, certPath string, registry api.RegistryCompose)
...
+	} else {
+		tagged, ok := repo.(reference.Tagged)
+		if !ok || tagged.Tag() == "" {
```
Comment (superxi911) [inline docker/manager.go]: I think it need to return err when "ok==false"

File: docker/manager.go (hunk 1 above)
Comment (xiaomuqiao) [inline docker/manager.go]: I deliberately do not quit，i think it is not so serious

Comment (xiaomuqiao) [inline docker/manager.go]: But it can also return here

**Comment by xiaomuqiao** [comment]
this pr is from "from unknown repository", cloned it. i will submit again.

**Pull request closed without merging**